__init__.py

```diff
@@ -0,0 +1,333 @@
+        n (int): The size of the window.
+        window_units (str): The type of units to use ("characters", "tokens", "lines", "sentences", "spans").
+        alignment_mode (str): How character indices snap to token boundaries.
+        - "strict" (no snapping)
+        - "contract" (span of all tokens completely within the character span)
+        - "expand" (span of all tokens at least partially covered by the character span)
+
+    Returns:
+        A generator of window strings.
+
+    Note:
+        Window boundaries are snapped to token boundaries in the original doc.
+        "Contract" means that the window will contain all tokens completely
+        within the boundaries of `i:i + n`. "Expand" means that window will
+        contain all tokens partially withn those boundaries. Setting
+        `alignment_mode="strict"` in `doc.char_span()` is not advised
+        because it returns `None` for any string that cannot be aligned to
+        token boundaries. As a result, a slice method is used if you want
+        to simply cut the text strictly on `n` characters.
+    """
+    # TODO: We have to iterate through the input twice to get the boundaries.
+    if isinstance(input, list):
+        input_spans = [span.as_doc() for span in input]
+        boundaries = [(i, i + n) for i in range(len(input_spans))]
+        for start, end in boundaries:
+            yield Doc.from_docs(input_spans[start:end]).text.strip()
+    else:
+        if isinstance(input, str):
+            alignment_mode = "strict"
+            boundaries = [(i, i + n) for i in range(len(input))]
+        else:
+            boundaries = [(i, i + n) for i in range(len(input.text))]
+        if alignment_mode == "strict":
+            for start_char, end_char in boundaries:
+                span = input[start_char:end_char]
+                if span is not None:
+                    yield span.text
+        else:
+            for start_char, end_char in boundaries:
+                span = input.char_span(
+                    start_char, end_char, alignment_mode=alignment_mode
+                )
+                if span is not None:
+                    yield span.text
+
+
+def sliding_windows(
+    input: Union[List[spacy.tokens.span.Span], spacy.tokens.doc.Doc],
+    n: int = 1000,
+    window_units: str = "characters",
+    alignment_mode: str = "strict",
+) -> Iterator:
+    """Create the windows generator.
+
+    Args:
+        input (Union[List[spacy.tokens.span.Span], spacy.tokens.doc.Doc]): A spaCy doc or a list of spaCy spans.
+        n (int): The size of the window.
+        window_units (str): The type of units to use ("characters", "tokens", "lines", "sentences", "spans").
+        alignment_mode (str): How character indices snap to token boundaries.
+        - "strict" (no snapping)
+        - "contract" (span of all tokens completely within the character span)
+        - "expand" (span of all tokens at least partially covered by the character span)
```

Comment:
Pulling these and the units out into a more global set of valid choices might be useful as you use them in a number of different places.